pygmt/src/inset.py

```diff
@@ -0,0 +1,131 @@
+        mirror opposite of *refpoint*. Specify inset box attributes via
+        the ``box`` option [outline only].
+    box : str or bool
+        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\
+        *pen*]][**+p**\ [*pen*]][**+r**\ [*radius*]][**+s**\
+        [[*dx*/*dy*/][*shade*]]]
+
+        If passed ``True``, this draws a rectangular box around the map
+        inset using the default pen; specify a different pen
+        with **+p**\ *pen*. Add **+g**\ *fill* to fill the logo box
+        [Default is no fill].
+        Append **+c**\ *clearance*  where *clearance* is either
+        *gap*, *xgap*\ /\ *ygap*, or *lgap*\ /\ *rgap*\ /\ *bgap*\ /\
+        *tgap* where these items are uniform, separate in x- and
+        y-direction, or individual side spacings between logo and border.
+        Append **+i** to draw a secondary, inner border as well. We use a
+        uniform *gap* between borders of 2\ **p** and the default pen
+        unless other values are specified. Append **+r** to draw rounded
+        rectangular borders instead, with a 6\ **p** corner radius. You
+        can override this radius by appending another value. Append
+        **+s** to draw an offset background shaded region. Here, *dx*/*dy*
+        indicates the shift relative to the foreground frame
+        [4\ **p**/-4\ **p**] and *shade* sets the fill style to use for
+        shading [Default is gray50].
+    margin : int or str or list
+        This is clearance that is added around the inside of the inset.
+        Plotting will take place within the inner region only. The margins
+        can be a single value, a pair of values separated (for setting
+        separate horizontal and vertical margins), or the full set of four
+        margins (for setting separate left, right, bottom, and top
+        margins). When passing multiple values, it can be either a list or
+        a string with the values separated by forward
+        slashes [Default is no margins].
+    no_clip : bool
+        Do NOT clip features extruding outside map inset boundaries [Default
+        will clip].
+    {V}
+
+    Examples
+    --------
+    >>> import pygmt
+    >>>
+    >>> # Create the larger figure
+    >>> fig = pygmt.Figure()
+    >>> fig.coast(region="MG+r2", water="lightblue", shorelines="thin")
+    >>> # Use a "with" statement to initialize the inset context manager
+    >>> # Setting the position to top left and a width of 3.5 centimeters
+    >>> with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen"):
+    ...     # Map elements under the "with" statement are plotted in the inset
+    ...     fig.coast(
+    ...         region="g",
+    ...         projection="G47/-20/?c",
+    ...         land="gray",
+    ...         water="white",
+    ...         dcw="MG+gred",
+    ...     )
+    ...
+    >>> # Map elements outside the `with` block are plotted in the main figure
+    >>> fig.logo(position="jBR+o0.2c+w3c")
```

Comment:
I'm wondering what happens if we add `fig.show()` here.

Reply:
One more thought on the example. We already have the example in the gallery 
![image](https://user-images.githubusercontent.com/3974108/106962641-6912af00-670d-11eb-9d24-8dc917dec1dc.png)

Do we really want to have the "example" in the docstring? I feel that examples in docstrings only make more sense to other non-plotting functions.

Reply:
I'm fine with removing it, but I do like the idea of having a quick "how to" at the bottom of the page is a nice feature for a user to not have to navigate to the tutorial/gallery example.

Reply:
Maybe keep it. I think the syntax of `fig.inset` is special enough to deserve an inline docstring example.

Reply:
OK, then better to add `fig.show()` at the end.

Reply:
```suggestion
    >>> fig.logo(position="jBR+o0.2c+w3c")
    >>> fig.show()
    <IPython.core.display.Image object>
```